File: analysis/energymonth.py
from sunsynk.resource import Resource


class EnergyMonth(Resource):
    def __init__(self, data):
        self.Load = None
        self.data = data
        self.energy = self.data['infos']
        for item in self.energy:
            if item['label'] == "Load" or item['label'] == "Load Power Consumption":
                self.Load = item
            elif item['label'] == "PV" or item['label'] == "PV Generation":
                self.PV = item
            elif item['label'] == "Export" or item['label'] == "Sold electricity":
                self.Export = item
            elif item['label'] == "Import" or item['label'] == "Purchased electricity":
                self.Import = item

    def get_Load(self):
        return self.Load

    def get_PV(self):
        return self.PV

    def get_Export(self):
        return self.Export

    def get_Import(self):
        return self.Import

File: android/app/src/main/java/com/richw/sunsynk/analysis/EnergyMonth.kt
package com.richw.sunsynk.analysis

import com.google.gson.JsonObject

class EnergyMonth(private val data: JsonObject) {

    data class LabelData(val label: String, val records: List<Record>)
    data class Record(val time: String, val value: String)

    private var load: LabelData? = null
    private var pv: LabelData? = null
    private var export: LabelData? = null
    private var import: LabelData? = null

    init {
        val infos = data.getAsJsonArray("infos")
        for (item in infos) {
            val obj = item.asJsonObject
            val label = obj.get("label").asString
            val records = obj.getAsJsonArray("records").map { r ->
                val rec = r.asJsonObject
                Record(rec.get("time").asString, rec.get("value").asString)
            }
            val ld = LabelData(label, records)
            when (label) {
                "Load", "Load Power Consumption" -> load = ld
                "PV", "PV Generation" -> pv = ld
                "Export", "Sold electricity" -> export = ld
                "Import", "Purchased electricity" -> import = ld
            }
        }
    }

    fun getLoad() = load
    fun getPv() = pv
    fun getExport() = export
    fun getImport() = import
}
